assets/javascript/codeBlockSwitch.js

```diff
@@ -0,0 +1,52 @@
+function addBlockSwitches() {
+    $('.listingblock.primary, .sidebarblock.primary').each(function() {
+        primary = $(this);
```

Comment:
Nitpicking (but not really): this looks like you want to declare a local variable, but you're actually unintentionally making `primary` a global variable (this is also known as polluting the global scope). If you really want to declare a local variable (and it totally looks like that's the case here and elsewhere in this script), use `let` (or `var`, but `let` has the civilized meaning).

It's enough that all the functions declared in this script go into the global scope :-) (It seems to me that they could easily be declared inside the `$(document).ready` callback and everything would work just fine.)

Reply:
FWIW, this is not my code :). I kept it as close as possible as upstream and fixed it to make it work with jQuery and the other things I needed.

Reply:
@Ladicek btw, if you want to fix things, be my guest. I think we will need to tweak this thing a lot so we might as well improve it.

Reply:
Fair enough :-)

Reply:
Pushed a commit with this change. All functions are now declared within the on-ready callback, and all local variables are actually local variables.